controller/src/beerocks/master/son_management.cpp

```diff
@@ -1535,6 +1535,51 @@ void son_management::handle_bml_message(Socket *sd,
         }
         break;
     }
+    case beerocks_message::ACTION_BML_WIFI_CREDENTIALS_UPDATE_REQUEST: {
+        LOG(TRACE) << "ACTION_BML_WIFI_CREDENTIALS_UPDATE_REQUEST";
+        uint32_t ret = 0;
+        auto request =
+            beerocks_header
+                ->addClass<beerocks_message::cACTION_BML_WIFI_CREDENTIALS_UPDATE_REQUEST>();
+        if (request == nullptr) {
+            LOG(ERROR) << "addClass cACTION_BML_WIFI_CREDENTIALS_UPDATE_REQUEST failed";
+            return;
+        }
+
+        auto al_mac = network_utils::mac_to_string(request->params());
+
+        auto agent_mac = database.get_node_parent_ire(al_mac);
```

Comment:
That doesn't sound right... the al_mac should be the al_mac already, so no database lookup should be needed...

Reply:
Here we are checking database - if al-mac presents in the database we send AP_CONFIGURATION_RENEW_MESSAGE else print error message in the log.

Reply:
I just reviewed the code and indeed `get_node_parent_ire` should return the IRE mac address - in this case it should return the bridge mac, which should be equal to the al_mac.

I think we should instead use the `get_all_connected_ires` API:
```
auto ires = database.get_all_connected_ires();
if (ires.find(request->al_mac() != ires.end()) {
  // send renew
} else {
  // print error
}
```